dawn/src/interface/atlas_interface.hpp

```diff
@@ -102,14 +111,28 @@ class Field {
 
   T* data() { return atlas_field_.data(); }
   const T* data() const { return atlas_field_.data(); }
+
   int numElements() const { return atlas_field_.shape(0) * atlas_field_.shape(1); }
+  std::tuple<size_t, size_t> getShape() const {
+    return std::make_tuple(atlas_field_.shape(0), atlas_field_.shape(1));
+  }
 
+  Field() : atlas_field_(atlas::array::make_view<T, 2>(*atlas::array::Array::create<T>(0, 0))){};
   Field(atlas::array::ArrayView<T, 2> const& atlas_field) : atlas_field_(atlas_field) {}
 
 private:
   atlas::array::ArrayView<T, 2> atlas_field_;
 };
 
+template <typename T>
+Field<T> allocateFieldLike(atlasTag, const Field<T>& other) {
+  auto shape = other.getShape();
+  // leaky!
+  auto field = new atlas::Field("copy", atlas::array::DataType::real64(),
```

Comment:
I understand that we need to do this because we don't have a register (held by the stencil class) of the allocated fields (and also we don't have a class to represent an allocated field, but only views).
This shows the need for a restructuring of naive ico.